submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/submitter/request/SubmissionRequestIndexer.kt

```diff
@@ -22,19 +21,16 @@ class SubmissionRequestIndexer(
         val sub = request.submission
 
         logger.info { "${sub.accNo} ${sub.owner} Started indexing submission files" }
-
         val totalFiles = indexSubmissionFiles(sub)
-        requestService.updateRequestTotalFiles(accNo, version, totalFiles)
-        requestService.updateRequestStatus(accNo, version, INDEXED)
+        requestService.saveSubmissionRequest(request.withNewStatus(INDEXED, totalFiles))
 
         logger.info { "${sub.accNo} ${sub.owner} Finished indexing submission files" }
     }
 
     private fun indexSubmissionFiles(sub: ExtSubmission): Int {
-        val index = AtomicInteger()
         return extSerializationService
             .fileSequence(sub)
-            .map { SubmissionRequestFile(sub.accNo, sub.version, index.incrementAndGet(), it.filePath, it) }
+            .mapIndexed { idx, file -> SubmissionRequestFile(sub.accNo, sub.version, idx + 1, file.filePath, file) }
```

Comment:
Is this actually better? I thought it was better to have AtomicInteger for this kind of operations instead of making the operation everytime

Reply:
what I like of current approach is that is kotlin folks the ones that implement index logic and not ourself so I think is simpler to us as we do not mantain that part of the code. Said that IMO diference is minimun.